infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -99,24 +99,13 @@ public class SessionManager {
      * @param upgradeManager      upgrade management module
      */
     public SessionManager(@Nonnull TopologyDescriptor topology, CorfuRuntime corfuRuntime,
-                          ServerContext serverContext, LogReplicationUpgradeManager upgradeManager) {
+                          ServerContext serverContext, LogReplicationUpgradeManager upgradeManager,
+                          String localCorfuEndpoint) {
```

Comment:
did you consider constructing the localCorfuEndpoint here like before?  As we already have the serverContext, the plugin file path is available and can be given to SaasEndpointProvider.  It will also eliminate the need to pass the extra parameter here

Reply:
if its not saas, then we'll need to recompute the endpoint by `getVersionFormattedEndpointURL`.
we would have some duplicate boilerplate code